### PR TITLE
Improved Dailog box

### LINKED ISSUE #430: Inappropriate dialog box in downloading fragment
**Is it a ?**
 Bug report
## Bug Report
When a user want to cancel the downloading wiki article,a dialog box is popping up with white shaded in background.I attached a screenshot below.

### Step 2 : What is your environment ?
  * browser / app : ___app__
  * version of the software you use : _2.2 Build:53__
  * device / operating system : ___Android Version 4.4.2 ( ASUS_T00J)__
  
### Step 3: Describe the problem:

  * What do you obtain? A copy of the error message or a screenshot is always useful
screenshot_2018-02-16-16-35-54

  * When does this occur? 
When user cancel the ongoing download in downloading fragment.

  * What should be the behaviour from your point of view? How do you expect the service to work?
    The behaviour of dialog box should be simple one clearing the white background as seen in the screen shot.

**Do you want to work on this?**
Yes

@mhutti1 @kelson42   Your reviews please?

### PR DESCRIPTION
**Pull request by RajaVamsi11**
Fixes #430

Changed the alert dialog box in downloading fragment for removal of white shadowed part.

### CODE REVIEW

File: app/src/main/java/org/kiwix/kiwixmobile/downloader/DownloadFragment.java [hunk 1]
```diff
@@ -267,10 +267,10 @@ public View getView(int position, View convertView, ViewGroup parent) {
       stop.setOnClickListener(v -> {
         hasArtificiallyPaused = LibraryFragment.mService.downloadStatus.get(mKeys[position]) == DownloadService.PLAY;
         setPlayState(pause, position, DownloadService.PAUSE);
-        new AlertDialog.Builder(faActivity, dialogStyle())
-            .setTitle(R.string.confirm_stop_download_title)
-            .setMessage(R.string.confirm_stop_download_msg)
-            .setPositiveButton(R.string.yes, (dialog, i) -> {
+        AlertDialog.Builder alertDialogBuilder=new AlertDialog.Builder(faActivity,dialogStyle());
```
Comment (srv-twry) [inline app/src/main/java/org/kiwix/kiwixmobile/downloader/DownloadFragment.java]: Add proper spaces before and after the ```=```

File: app/src/main/java/org/kiwix/kiwixmobile/downloader/DownloadFragment.java [hunk 2]
```diff
@@ -280,13 +280,14 @@ public View getView(int position, View convertView, ViewGroup parent) {
                 zimManageActivity.mSectionsPagerAdapter.libraryFragment.libraryAdapter.getFilter().filter(((ZimManageActivity) getActivity()).searchView.getQuery());
               }
             })
-            .setNegativeButton(R.string.no, (dialog, i) -> {
-              if (hasArtificiallyPaused) {
+                         .setNegativeButton(R.string.no, (dialogInterface, i) -> {
```
Comment (srv-twry) [inline app/src/main/java/org/kiwix/kiwixmobile/downloader/DownloadFragment.java]: wrong indentation.

File: app/src/main/java/org/kiwix/kiwixmobile/downloader/DownloadFragment.java [hunk 3]
```diff
@@ -280,13 +280,14 @@ public View getView(int position, View convertView, ViewGroup parent) {
...
+                         if (hasArtificiallyPaused) {
                 hasArtificiallyPaused = false;
                 setPlayState(pause, position, DownloadService.PLAY);
               }
-            })
-            .show();
+            });
+        AlertDialog alertDialog = alertDialogBuilder.create();
```
Comment (srv-twry) [inline app/src/main/java/org/kiwix/kiwixmobile/downloader/DownloadFragment.java]: Are we actually using this ```alertDialog``` variable except in the next line ? If not, then replace the lines 289 and 290 with 
``` alertDialogBuilder.create().show()```

File: app/src/main/java/org/kiwix/kiwixmobile/downloader/DownloadFragment.java (hunk 1 above)
Comment (RajaVamsi11) [reply app/src/main/java/org/kiwix/kiwixmobile/downloader/DownloadFragment.java]: Ok @srv-twry

File: app/src/main/java/org/kiwix/kiwixmobile/downloader/DownloadFragment.java (hunk 3 above)
Comment (RajaVamsi11) [reply app/src/main/java/org/kiwix/kiwixmobile/downloader/DownloadFragment.java]: Yeah ..we can do that.I will change it

**Comment by RajaVamsi11** [comment]
@srv-twry  Take a look

File: app/src/main/java/org/kiwix/kiwixmobile/downloader/DownloadFragment.java [hunk 4]
```diff
@@ -267,26 +267,27 @@ public View getView(int position, View convertView, ViewGroup parent) {
...
+        AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(faActivity,dialogStyle());
+        alertDialogBuilder
+                .setTitle(R.string.confirm_stop_download_title)
+                .setMessage(R.string.confirm_stop_download_msg)
+                .setPositiveButton(R.string.yes,(dialogInterface, i) -> {
+                    LibraryFragment.mService.stopDownload(mKeys[position]);
+                    mDownloads.remove(mKeys[position]);
+                    mDownloadFiles.remove(mKeys[position]);
+                    downloadAdapter.notifyDataSetChanged();
+                                 updateNoDownloads();
```
Comment (srv-twry) [inline app/src/main/java/org/kiwix/kiwixmobile/downloader/DownloadFragment.java]: Wrong indentation here.

File: app/src/main/java/org/kiwix/kiwixmobile/downloader/DownloadFragment.java [hunk 5]
```diff
@@ -267,26 +267,27 @@ public View getView(int position, View convertView, ViewGroup parent) {
...
               if (zimManageActivity.mSectionsPagerAdapter.libraryFragment.libraryAdapter != null) {
                 zimManageActivity.mSectionsPagerAdapter.libraryFragment.libraryAdapter.getFilter().filter(((ZimManageActivity) getActivity()).searchView.getQuery());
               }
             })
-            .setNegativeButton(R.string.no, (dialog, i) -> {
-              if (hasArtificiallyPaused) {
+                .setNegativeButton(R.string.no, (dialogInterface, i) -> {
```
Comment (srv-twry) [inline app/src/main/java/org/kiwix/kiwixmobile/downloader/DownloadFragment.java]: Wrong indentation again. It should be alligned with the ```.setPositiveButton```

File: app/src/main/java/org/kiwix/kiwixmobile/downloader/DownloadFragment.java (hunk 2 above)
Comment (RajaVamsi11) [reply app/src/main/java/org/kiwix/kiwixmobile/downloader/DownloadFragment.java]: .setNegativeButton(.................)  is exactly under the 
.setPositiveButton(..................)

File: app/src/main/java/org/kiwix/kiwixmobile/downloader/DownloadFragment.java (hunk 5 above)
Comment (RajaVamsi11) [reply app/src/main/java/org/kiwix/kiwixmobile/downloader/DownloadFragment.java]: It is alligned with .setPositiveButton

File: app/src/main/java/org/kiwix/kiwixmobile/downloader/DownloadFragment.java [hunk 6]
```diff
@@ -267,26 +267,27 @@ public View getView(int position, View convertView, ViewGroup parent) {
...
+                    LibraryFragment.mService.stopDownload(mKeys[position]);
+                    mDownloads.remove(mKeys[position]);
+                    mDownloadFiles.remove(mKeys[position]);
+                    downloadAdapter.notifyDataSetChanged();
+                    updateNoDownloads();
+                    if (zimManageActivity.mSectionsPagerAdapter.libraryFragment.libraryAdapter != null) {
+                        zimManageActivity.mSectionsPagerAdapter.libraryFragment.libraryAdapter.getFilter().filter(((ZimManageActivity) getActivity()).searchView.getQuery());
+                     }
+                })
+                .setNegativeButton(R.string.no, (dialogInterface, i) -> {
```
Comment (RajaVamsi11) [inline app/src/main/java/org/kiwix/kiwixmobile/downloader/DownloadFragment.java]: .setNegativeButton is alligned to
.setPositive Button

File: app/src/main/java/org/kiwix/kiwixmobile/downloader/DownloadFragment.java (hunk 5 above)
Comment (srv-twry) [reply app/src/main/java/org/kiwix/kiwixmobile/downloader/DownloadFragment.java]: Yes :neutral_face: , but actually the lines above it weren't , hence i got confused.

File: app/src/main/java/org/kiwix/kiwixmobile/downloader/DownloadFragment.java (hunk 6 above)
Comment (RajaVamsi11) [reply app/src/main/java/org/kiwix/kiwixmobile/downloader/DownloadFragment.java]: Took special care :-)

Comment (srv-twry) [reply app/src/main/java/org/kiwix/kiwixmobile/downloader/DownloadFragment.java]: Not a problem. As long as you learn from them and don't make them again :smile:

**Comment by RajaVamsi11** [comment]
@srv-twry  Done!!!

**Comment by srv-twry** [comment]
@mhutti1 I apologize, It's weird though. I had downloaded it locally and checked and it worked so i approved the PR.
Here is the screenshot:
screenshot_2018-02-18-05-15-41-23

I am not getting the error that's why approved this PR. But yeah you're right, it doesn't look like any semantic changes have been made in this PR. I'll pay more attention from next time.

**Comment by srv-twry** [comment]
Probably the issue itself was platform specific and hence wasn't there at the first place on my phone.

**Comment by RajaVamsi11** [comment]
The problem is in outdated version of the app.I think that problem has been solved in the present version of the app.So i am closing this.

**Pull request closed without merging**